external/flume-assembly/pom.xml

```diff
@@ -0,0 +1,135 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-parent_2.10</artifactId>
+    <version>1.5.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <groupId>org.apache.spark</groupId>
+  <artifactId>spark-streaming-flume-assembly_2.10</artifactId>
```

Comment:
Nope, none of the fume stuff is present in the spark-assembly. That is
precisely why this assembly JAR with spark-streaming-flume and flume+its
dependencies were generated.

On Mon, Jul 6, 2015 at 2:02 PM, Marcelo Vanzin <notifications@github.com>
wrote:

> In external/flume-assembly/pom.xml
> <https://github.com/apache/spark/pull/6830#discussion_r33981901>:
>
> > +  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +  ~ See the License for the specific language governing permissions and
> > +  ~ limitations under the License.
> > +  -->
> > +
> > +<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
> > +  <modelVersion>4.0.0</modelVersion>
> > +  <parent>
> > +    <groupId>org.apache.spark</groupId>
> > +    <artifactId>spark-parent_2.10</artifactId>
> > +    <version>1.5.0-SNAPSHOT</version>
> > +    <relativePath>../../pom.xml</relativePath>
> > +  </parent>
> > +
> > +  <groupId>org.apache.spark</groupId>
> > +  <artifactId>spark-streaming-flume-assembly_2.10</artifactId>
>
> Hi guys,
>
> Sorry I'm late to the party, but why is this new assembly necessary?
>
> It creates an 80MB jar file that repackages a bunch of things already
> present in the Spark assembly (e.g. scala.*, org.hadoop.*, and a whole
> lot of other things). If python/pyspark/streaming/flume.py is meant to be
> used inside a Spark application, aren't those dependencies already provided
> by the Spark assembly? In which case all that is needed is the existing
> spark-streaming-flume artifact?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/6830/files#r33981901>.
>